File: algorithms/implementation/python3/chocolate_feast.py
#!/bin/python3

import sys


t = int(input().strip())
for a0 in range(t):
    n, c, m = input().strip().split(' ')
    n, c, m = [int(n),int(c),int(m)]
    
    total_chocolates_bought = n // c    
    chocolates_bought = total_chocolates_bought

    wrappers_left = 0
    while(chocolates_bought >= m):
        total_chocolates_bought += (chocolates_bought // m)
        wrappers_left = chocolates_bought % m
        chocolates_bought = (chocolates_bought // m) + wrappers_left
    
    print(total_chocolates_bought)

File: algorithms/implementation/C#/ChocolateFeast.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {

    static void Main(String[] args) {
        int t = Convert.ToInt32(Console.ReadLine());
        for(int a0 = 0; a0 < t; a0++){
            string[] tokens_n = Console.ReadLine().Split(' ');
            int n = Convert.ToInt32(tokens_n[0]);
            int c = Convert.ToInt32(tokens_n[1]);
            int m = Convert.ToInt32(tokens_n[2]);
            
            int totalChocolatesBought = n / c;
            int chocolatesBought = totalChocolatesBought;

            while(chocolatesBought >= m)
            {               
                totalChocolatesBought += (chocolatesBought / m);
                int wrappersLeft = chocolatesBought % m;
                chocolatesBought = (chocolatesBought / m) + wrappersLeft;  
            }
            Console.WriteLine(totalChocolatesBought);
        }
    }
}
